execution/gethexec/executionengine.go

```diff
@@ -654,17 +721,58 @@ func (s *ExecutionEngine) getL1PricingSurplus() (int64, error) {
 	return surplus.Int64(), nil
 }
 
-func (s *ExecutionEngine) cacheL1PriceDataOfMsg(num arbutil.MessageIndex, receipts types.Receipts, block *types.Block) {
+func (s *ExecutionEngine) cacheL1PriceDataOfMsg(seqNum arbutil.MessageIndex, receipts types.Receipts, block *types.Block) {
```

Comment:
This function is currently only called from sequenceTransactionsWithBlockMutex - but it (or equivalent) should also be called from anything that adds a block. Possibly, it'll be easier from appendBlock. However, need to make sure that delayed messages don't count towards callDataUnits (gasUsedForL1 should already be 0 for them)

When we do that - resetCache() should be quite unexpected and only happen in case of a reorg
I also think the same function should update L1GasEstimate

Reply:
Done.